assets/src/models/detoursList.ts

```diff
@@ -0,0 +1,59 @@
+import { array, Infer, nullable, number, string, type } from "superstruct"
+
+export const SimpleDetour = type({
+  route: string(),
+  direction: string(),
+  name: string(),
+  intersection: string(),
+  updatedAt: number(),
+})
+
+export type SimpleDetour = Infer<typeof SimpleDetour>
+
+export const SimpleDetourFromApi = type({
+  route: string(),
+  direction: string(),
+  name: string(),
+  intersection: string(),
+  updated_at: number(),
+})
+
+export type SimpleDetourFromApi = Infer<typeof SimpleDetourFromApi>
+
+export const simpleDetourFromData = (
+  detourData: SimpleDetourFromApi
+): SimpleDetour => ({
+  route: detourData.route,
+  direction: detourData.direction,
+  name: detourData.name,
+  intersection: detourData.intersection,
+  updatedAt: detourData.updated_at,
+})
+
+export const GroupedSimpleDetours = type({
+  active: nullable(array(SimpleDetour)),
+  draft: nullable(array(SimpleDetour)),
+  past: nullable(array(SimpleDetour)),
+})
+
+export type GroupedSimpleDetours = Infer<typeof GroupedSimpleDetours>
+
+export const GroupedDetoursFromApi = type({
+  active: nullable(array(SimpleDetourFromApi)),
+  draft: nullable(array(SimpleDetourFromApi)),
+  past: nullable(array(SimpleDetourFromApi)),
+})
+
+export type GroupedDetoursFromApi = Infer<typeof GroupedDetoursFromApi>
+
+export const groupedDetoursFromData = (
+  groupedDetours: GroupedDetoursFromApi
+): GroupedSimpleDetours => ({
+  active:
+    groupedDetours.active?.map((detour) => simpleDetourFromData(detour)) ||
```

Comment:
Question (non-blocking): Is the `|| null` necessary?

Reply:
Ope forgot to check. I think it's necessary; if `active?` is not found, it will return `undefined`, and the type expects a `null`. So removing the `|| null` results in: `Type 'undefined' is not assignable to type '{ route: string; direction: string; name: string; intersection: string; updatedAt: number; }[] | null'.`

Reply:
Huh! You learn something new every day! My intuition told me that `{ a: null }.a?.nested` would return `null`, but yeah it returns `undefined`!

Reply:
Since I refactored the models, I did tweak this a bit! Let me know what you think